emgapi/metagenomics_exchange.py

```diff
@@ -0,0 +1,161 @@
+        self.broker = settings.METAGENOMICS_EXCHANGE_MGNIFY_BROKER
+
+    def get_request(self, endpoint: str, params: dict):
+        """Make a GET request, returns the response"""
+        headers = {"Accept": "application/json", "Authorization": self.__token}
+        response = requests.get(
+            f"{self.base_url}/{endpoint}", headers=headers, params=params
+        )
+        response.raise_for_status()
+        return response
+
+    def post_request(self, endpoint: str, data: dict):
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Authorization": self.__token,
+        }
+        response = requests.post(
+            f"{self.base_url}/{endpoint}", json=data, headers=headers
+        )
+        response.raise_for_status()
+        return response
+
+    def delete_request(self, endpoint: str):
+        headers = {
+            "Accept": "application/json",
+            "Authorization": self.__token,
+        }
+        response = requests.delete(
+            f"{self.base_url}/{endpoint}", headers=headers
+        )
+        response.raise_for_status()
+        return response
+
+    def patch_request(self, endpoint: str, data: dict):
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Authorization": self.__token,
+        }
+        response = requests.patch(
+            f"{self.base_url}/{endpoint}", json=data, headers=headers
+        )
+        response.raise_for_status()
+        return response
+
+    def generate_metadata(self, mgya, run_accession, status):
+        return {
+            "confidence": "full",
+            "endPoint": f"https://www.ebi.ac.uk/metagenomics/analyses/{mgya}",
+            "method": ["other_metadata"],
+            "sourceID": mgya,
+            "sequenceID": run_accession,
+            "status": status,
+            "brokerID": self.broker,
+        }
+
+    def add_analysis(self, mgya: str, run_accession: str, public: bool):
+        data = self.generate_metadata(mgya, run_accession, public)
+        response = self.post_request(endpoint="datasets", data=data)
+        return response
+
+
+    def check_analysis(self, source_id: str, sequence_id: str, public=None, metadata=None) -> [str, bool]:
+        logging.info(f"Check {source_id} {sequence_id}")
+        params = {}
+        if public:
+            params = {
+                "status": "public" if public else "private",
+                "broker": self.broker
+            }
+        endpoint = f"sequences/{sequence_id}/datasets"
+        analysis_registry_id = None
+        metadata_match = True
+
+        try:
+            response = self.get_request(endpoint=endpoint, params=params)
+        except:
+            logging.warning(f"Get API request failed")
+            return analysis_registry_id , metadata_match
+
+        if response.ok:
```

Comment:
I think it is not necessary to check if response.ok here. 
If the response is ok, it should fall in the try block. 
Otherwise it would fall in the except block.